### PR TITLE
Feature/gh 11232 migration of feature keys

### LINKED ISSUE #11232: Migrate old featureModules config keys to new ones
See https://github.com/SAP/spartacus/issues/11231

We should drop default config for old keys and provide schematics migration (at least adding a code comment in customers code near the old keys).

### PR DESCRIPTION
**Pull request by ChristophHi**
Adds comments in case customers use the obsolete feature keys for the rulebased and textfield configurators.

Closes #11232

### CODE REVIEW

File: projects/schematics/src/migrations/4_0/product-configurator-changed-feature-keys/product-configurator-changed-feature-keys.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  function addCommentToFeatureKey(
+    tree: Tree,
+    mainPath: any,
+    obsoleteKey: string,
+    comment: string
+  ) {
+    const appModulePath = getAppModulePath(tree, mainPath);
+    const appModuleSource = getTsSourceFile(tree, appModulePath);
+    const changes = insertCommentAboveConfigProperty(
```
Comment (Platonn) [inline projects/schematics/src/migrations/4_0/product-configurator-changed-feature-keys/product-configurator-changed-feature-keys.ts]: Do we really need this this custom schematics file? Did you consider reusing the generic mechanism /config-deprecations? You might want to copy-paste it from folder /migrations/3_0.

Comment (ChristophHi) [reply projects/schematics/src/migrations/4_0/product-configurator-changed-feature-keys/product-configurator-changed-feature-keys.ts]: Hmhh yes you are probably right. I thought that there would be too many matches when searching for textfield/rulebased in _all_ configurations but actually it only appears in customers app.module.ts

Comment (ChristophHi) [reply projects/schematics/src/migrations/4_0/product-configurator-changed-feature-keys/product-configurator-changed-feature-keys.ts]: I did those changes and unit tested them

File: docs/migration/4_0.md [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+      }
+
+needs to look like that in 4.0
+
+      featureModules: {
+        productConfiguratorRulebased: {
+          module: () => import('@spartacus/product-configurator/rulebased').then(
+          (m) => m.RulebasedConfiguratorModule
+        ),
+        },
```
Comment (Platonn) [inline docs/migration/4_0.md:28]: nitpick: for better .md formatting, you might want to wrap code snippets in 
<pre>
```typescript
  code here
```
</pre>

File: docs/migration/4_0.md [hunk 3]
```diff
@@ -1,3 +1,29 @@
...
+      }
```
Comment (Platonn) [inline docs/migration/4_0.md]: nitpick: for better .md formatting, you might want to wrap code snippets in 
<pre>
```typescript
  code here
```
</pre>

**Review by Platonn — DISMISSED** [review]

**Review by Platonn — APPROVED** [review]

**Pull request merged**